### DISCUSSION TITLE
Push Notifications Android

### DISCUSSION THREAD

**Discussion by mikescola** [buildnatively/community#78] [Ideas]
### Please confirm you've checked the following resources:

- [X] I've searched through the [Docs](https://docs.buildnatively.com).
- [X] I've checked [Natively Youtube Playlist 1](https://youtube.com/playlist?list=PLYo9EJqlVMNdtzb_rm6_6kjlDCmaoAqlO&si=sOygZtRDGUn_2Yi8) and [Natively Youtube Playlist 2](https://youtube.com/playlist?list=PLYo9EJqlVMNcwz2o1Xx5xVYYzzKgcyMeB&si=f4GuCPvUAdjvHplD).
- [X] I've reviewed the [Troubleshooting Guides](https://docs.buildnatively.com/guides/troubleshooting).
- [X] I've visited the [FAQ](https://docs.buildnatively.com/natively-platform/faq).

### Which web app platform are you using?

Other

### Which SDK are you using?

JavaScript SDK

### JS SDK / Bubble Plugin Version (e.g., 2.12.1)

2.12.3

### Platform with the issue?

Android

### App ID

1688140955740x429467044870357000

### Steps to reproduce

I'm unsure how to actually reproduce this. Basically, the app is installed and push notifications are registered. These work for a period of time, and then they just "stop working". OneSignal shows as delivered and everything seems correct, I believe this to be an issue in the actual app build itself. This does not happen in the iOS build.

### Expected results

Push notifications continue to come unless revoked on the Android platform.

### Actual results

Push notifications stop working after a period of time on the Android platform.

**Comment by romanfurman6** [comment]
This is something Android-related issue. Checkout these video: https://youtu.be/VtHHZeNKmXE?t=577

**Reply by mikescola** [reply]
Hey Roman! So what the video is outlining is push notifications not happening at first on launch (I believe), but this will start happening after we receive successful push notifications. Like my personal application for example no longer receives them, the only way to fix it is to uninstall the app and reinstall. Once I do this I will get push notifications for a few days and then they will stop again.

**Reply by romanfurman6** [reply]
Can you double check if player_id is correct

**Reply by romanfurman6** [reply]
i mean maybe at some point you set to a user a different player_id

**Reply by romanfurman6** [reply]
I think install/uninstall can be potentially cause the user to login one more time (and you're set a player id)
player_id change only on install/uninstall

**Reply by romanfurman6** [reply]
did you saw a similiar issue on iOS ?

**Reply by mikescola** [reply]
No, I have not seen this issue on iOS. My application accounts for the possibility of multiple player_ids associated with a single user, so that's not the issue. I always make sure the data is in sync.

**Reply by romanfurman6** [reply]
That's kind of weird... Can you please check this page: https://documentation.onesignal.com/docs/notifications-show-successful-but-are-not-being-shown

**Reply by romanfurman6** [reply]
Hello, @mikescola did you had a chance to solve this?

**Reply by mikescola** [reply]
I've gone through all the documentation and I can't get it to resolve. Everything is setup correctly and used to work with no problems. It definitely is a regression but I can't pinpoint to you where it is at.

**Reply by romanfurman6** [reply]
@mikescola Hmm, let's mb jump on a call https://link.buildnatively.com/help
